services/blockchain-indexer/shared/dataService/business/events.js

```diff
@@ -70,6 +71,31 @@ const getEventsByHeight = async (height) => {
 	return eventsFromNode;
 };
 
+const getEventsByBlockID = async (blockID) => {
+	// Get from cache
+	const cachedEvents = await eventCacheByBlockID.get(blockID);
+	if (cachedEvents) return JSON.parse(cachedEvents);
+
+	// Get from DB only when isPersistEvents is enabled
+	if (config.isPersistEvents) {
+		const eventsTable = await getEventsTable();
+		const dbEventStrs = await eventsTable.find({ blockID }, ['eventStr']);
+
+		if (dbEventStrs.length) {
+			const dbEvents = dbEventStrs
+				.map(({ eventStr }) => eventStr ? JSON.parse(eventStr) : eventStr);
+			await eventCacheByBlockID.set(blockID, JSON.stringify(dbEvents));
```

Comment:
Caching of the data can be done asynchronously without delaying the response.
```suggestion
			eventCacheByBlockID.set(blockID, JSON.stringify(dbEvents));
```